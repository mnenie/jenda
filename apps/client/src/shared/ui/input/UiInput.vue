<script setup lang="ts">
import { useVModel } from '@vueuse/core'
import type { HTMLAttributes } from 'vue'
import { cn } from '@/shared/libs/shadcn/utils'

const props = defineProps<{
  defaultValue?: string | number
  modelValue?: string | number
  class?: HTMLAttributes['class']
}>()

const emits = defineEmits<{
  (e: 'update:modelValue', payload: string | number): void
}>()

const modelValue = useVModel(props, 'modelValue', emits, {
  passive: true,
  defaultValue: props.defaultValue,
})
</script>

<template>
  <input
    v-model="modelValue"
    :class="cn(
      'input',
      props.class,
    )"
  />
</template>
