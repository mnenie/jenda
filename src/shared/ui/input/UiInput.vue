<script setup lang="ts">
import type { InputTypeHTMLAttribute } from 'vue';

type InputProps = {
  modelValue?: string | number;
  placeholder?: string;
  type?: InputTypeHTMLAttribute;
};

withDefaults(defineProps<InputProps>(), {
  modelValue: '',
  placeholder: 'dizzo',
  type: 'text'
});

const emit = defineEmits(['update:modelValue']);
</script>

<template>
  <input
    :value="modelValue"
    :placeholder="placeholder"
    :type="type"
    :class="[$style.input_default, 'text-sm']"
    @input="emit('update:modelValue', ($event.target as HTMLInputElement).value)"
  />
</template>

<style module lang="scss">
@import '@/app/styles/mixins';

.input_default {
  display: flex;
  width: 100%;
  height: 32px;
  border: 1px solid var(--zinc-200);
  border-radius: 6px;
  background-color: transparent;
  padding: 4px 12px;
  &::placeholder {
    color: var(--zinc-500);
  }
  @include transition;
  @include on-focus {
    outline: none;
    border: 1px solid var(--zinc-800);
  }
}
</style>
