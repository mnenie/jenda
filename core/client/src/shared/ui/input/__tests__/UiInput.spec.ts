import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import UiInput from '../UiInput.vue'

describe('uiInput', () => {
  const wrapper = shallowMount(UiInput, {
    props: {
      modelValue: 'value',
      placeholder: 'jenda',
      type: 'text',
    },
  })

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should update value with @input', async () => {
    const input = wrapper.find('input')
    await input.setValue('new')
    expect(wrapper.emitted('update:modelValue')![0]).toEqual(['new'])
  })
})
