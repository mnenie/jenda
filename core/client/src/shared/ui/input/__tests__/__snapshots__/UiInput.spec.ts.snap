// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`uiInput > should render correctly 1`] = `"<input placeholder="jenda" type="text" class="text-sm input_default" autocomplete="on" value="value">"`;
